src/auth/auth-provider.tsx

```diff
@@ -61,78 +60,40 @@ export function AuthProvider({
 export type AuthenticationPayload = {
   username: string
   id: number
-  token: string
-  refreshToken(usedToken: string): Promise<void>
-  clearToken(): void
 } | null
 
-function useAuthentication(): [RefObject<AuthenticationPayload>, boolean] {
-  const initialValue = parseAuthCookie()
-  const authenticationPayload = useRef<AuthenticationPayload>(initialValue)
-  const pendingRefreshTokenPromise = useRef<Promise<void> | null>(null)
-
-  const [loggedIn, setLoggedIn] = useState(() => {
-    return initialValue !== null
-  })
-
-  function parseAuthCookie(): AuthenticationPayload {
-    try {
-      const cookies = typeof window === 'undefined' ? {} : Cookies.get()
-
-      const { access_token, id_token } = JSON.parse(
-        cookies['auth-token']
-      ) as Token
-
-      const decoded = jwt_decode<{
-        username: string
-        id: number
-      }>(id_token as string)
-
-      return {
-        username: decoded.username,
-        id: decoded.id,
-        token: access_token as string,
-        refreshToken,
-        clearToken,
-      }
-    } catch {
-      return null
+function parseAuthCookie(session: Session): AuthenticationPayload {
+  try {
+    const username = (session?.identity?.traits as { username: string })
+      ?.username
+    const legacyId = (
+      session?.identity?.metadata_public as { legacy_id: number }
+    )?.legacy_id
+    return {
+      username,
+      id: legacyId,
     }
+  } catch {
+    return null
   }
+}
 
-  // eslint-disable-next-line @typescript-eslint/require-await
-  async function refreshToken(usedToken: string) {
-    async function startRefreshTokenPromise(): Promise<void> {
-      if (typeof window === 'undefined') return
-
-      await fetch('/api/auth/refresh-token', {
-        method: 'POST',
-      })
-
-      authenticationPayload.current = parseAuthCookie()
-      pendingRefreshTokenPromise.current = null
-    }
-
-    const currentCookieValue = parseAuthCookie()
-    if (currentCookieValue === null || currentCookieValue.token !== usedToken) {
-      // Cooke has a newer token than the one we used for the request. So use that instead.
-      authenticationPayload.current = currentCookieValue
-      return
-    }
-
-    if (!pendingRefreshTokenPromise.current) {
-      // Only initiate the token refresh request if it has not been started already.
-      pendingRefreshTokenPromise.current = startRefreshTokenPromise()
-    }
+function useAuthentication(): [RefObject<AuthenticationPayload>, boolean] {
+  const [session, setSession] = useState<Session | null>(null)
+  const authenticationPayload = useRef<AuthenticationPayload>(null)
 
-    return pendingRefreshTokenPromise.current
-  }
+  useEffect(() => {
+    void (async () => {
+      const { data } = await kratos.toSession().catch()
```

Comment:
The `.catch()` looks weird and might really catch all potential errors without reporting them.

Reply:
Sorry, actually, it is not good as implemented in https://github.com/serlo/frontend/commit/cbee43211ef4fb5812e603cb133670969edeb8f9 because it keeps logging into the console `GEThttp://localhost:3000/api/.ory/sessions/whoami [HTTP/1.1 401 Unauthorized 700ms]`.
We have to have the catch after toSession(), but I got your idea.